test/actions.test.ts

```diff
@@ -2,68 +2,109 @@ import * as chai from "chai";
 import * as chaiAsPromised from "chai-as-promised"
 import * as dotenv from "dotenv";
 import { ethers } from "ethers";
+import { createInstance } from "../src";
 
 import * as actions from "../src/actions";
-import { getSaleStatus, getWhitelist } from "../src/actions";
-import { getWhiteListSaleContract } from "../src/contracts";
-import { Claim, IPFSGatewayUri, SaleStatus } from "../src/types";
+import { getSaleStatus, getMintlist } from "../src/actions";
+import { getWolfSaleContract, WolfSale } from "../src/contracts";
+import { Claim, Config, Instance, Maybe, Mintlist, SaleStatus } from "../src/types";
 
 const expect = chai.expect;
 chai.use(chaiAsPromised.default);
 dotenv.config();
 
 describe("Test Custom SDK Logic", () => {
-  // Kovan
-  const whitelistSimpleSaleContractAddress =
-    "0xa6A3321b743C31912263090275E24d8b1A50cFE8";
   const provider = new ethers.providers.JsonRpcProvider(
-    process.env["INFURA_URL"]
+    process.env["INFURA_URL"],
+    4
   );
-  const pk = process.env["PRIVATE_KEY"];
+
+  const pk = process.env["ASTRO_PRIVATE_KEY"];
   if (!pk) throw Error("No private key");
 
   const signer = new ethers.Wallet(pk, provider);
 
   const voidSignerAddress = "0x8ba1f109551bD432803012645Ac136ddd64DBA72";
-  const addressFromFile = "0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65";
+  const astroTest = "0x35888AD3f1C0b39244Bb54746B96Ee84A5d97a53";
 
   // Sample merkle file
   const ipfsUrl =
     "ipfs://bafybeihy6gypowkcfwtsuvspbqah6chvtfjd3lz5xcjdvwnwyjdxh2bufa";
 
-  describe("getWhitelistedUserClaim", () => {
+  // From dApp Rinkeby zSale SDK Config
+  const config: Config = {
+    web3Provider: provider,
+    contractAddress: "0xC82E9E9B1e28F10a4C13a915a0BDCD4Db00d086d",
+    // Using Rinkeby not Kovan, but can use same file for tests
+    merkleTreeFileUri: 'https://d3810nvssqir6b.cloudfront.net/kovan-test-merkleTree.json',
+    advanced: {
+      merkleTreeFileIPFSHash:
+        'Qmf8XuYT181zdvhNXSeYUhkptgezzK8QJnrAD16GGj8TrV',
+    },
+  }
+
+  describe("e2e purchase", () => {
+    it("runs with access list", async () => {
+      const sdk: Instance = createInstance(config);
+      const mintlist = await sdk.getMintlist()
+
+      const wolfSale = await getWolfSaleContract(signer, config.contractAddress)
+      const address = await signer.getAddress()
+
+      const args = {
+        count: ethers.BigNumber.from("1"),
```

Comment:
You could just do `1` instead of `BigNumber.from`

Reply:
Actually `sdk..purchaseDomains(...)` first parameter is `ethers.BigNumber` not `ethers.BigNumberish` so it has to be this, or I can change the param but that complicates the error checking because now `count` can by any of 5 different types.

Reply:
Ah, that makes sense. I think ideally we would've used a plain old `number` since `sdk.purchaseDomains` is a user-facing function and they don't need the space that `BigNumber` provides. But since it's already a `BigNumber` let's just keep it that way since it will be a breaking change to users if we change it now.